### PR TITLE
Add monthly pricing toggle to product types

### LINKED ISSUE #4741: Onboarding - add a price toggle to the product selection step and move extension details to a popover
I've done a little exploration around ways we might display extension prices on a monthly basis, but still allow folks to view them annually as well. This will hopefully strike a better balance between transparency and value communication.

It also avoids some unnecessary repetition, and moving the extension details to a popover saves a lot of visual space.

### Before

before

### After

design

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #4741

Adds a monthly pricing toggle next to product types as well as popovers for product type descriptions.

### Screenshots
Screen Shot 2020-08-19 at 5 16 55 PM

### Detailed test instructions:

1. Check that the popovers correctly display the product description and learn more link when clicked.
1. Check that the learn more link still fires the appropriate event.
1. Check that toggling the pricing between monthly and annual shows the correct prices and pill text.
1. Check that style matches expected.

/cc @jameskoster I mostly kept styling consistent with other checkbox groups, but made some stabs at the colors and font-sizes.  Feel free to jump and in and add things or let me know what you'd like to see changed.

### CODE REVIEW

**Review by jameskoster — APPROVED** [review]
Opened a PR with adjustments for this (and some other things - sorry) [here](https://github.com/woocommerce/woocommerce-admin/pull/5019).

Conditionally approving based on that PR being merged.

**Review by timmyc — COMMENTED** [review]
This is looking good Josh, just a couple of questions.

File: client/profile-wizard/steps/product-types/label.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+				</Popover>
+			) }
+			<Tooltip text={ toolTipText } position="bottom center">
+				<Pill>
+					<span className="screen-reader-text">{ toolTipText }</span>
+					{ isMonthlyPricing
+						? sprintf(
+								/* translators: Dollar amount (example: $4.08 ) */
+								__( '$%f per month', 'woocommerce-admin' ),
+								( annualPrice / 12.0 ).toFixed( 2 )
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/profile-wizard/steps/product-types/label.js:94]: This was one of my original concerns and reasons for providing a higher estimate on this issue as well, but I think the REST response for product types will always be in USD so it should not be converted to local currency format.

File: client/profile-wizard/steps/product-types/label.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import { Fragment, useState } from '@wordpress/element';
+import interpolateComponents from 'interpolate-components';
+import { Link, Pill } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { recordEvent } from '../../../lib/tracks';
+
+export default function ProductTypeLabel( {
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/profile-wizard/steps/product-types/label.js:15]: Great idea.  Will get some added today.

File: client/profile-wizard/steps/product-types/test/index.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	} );
+
+	test( 'should show validation error on continue', async () => {
+		const mockGoToNextStep = jest.fn();
+		const mockUpdateProfileItems = jest.fn().mockResolvedValue();
+
+		render(
+			<ProductTypes
+				goToNextStep={ mockGoToNextStep }
+				updateProfileItems={ mockUpdateProfileItems }
```
Comment (joshuatf) [inline client/profile-wizard/steps/product-types/test/index.js:60]: This does not seem to be mocking properly.  Can someone on @woocommerce/isotope take a look at this?

**Comment by joshuatf** [comment]
@timmyc Added in those tests, but one is failing.  Need some feedback on that piece, but other tests are in place.

File: client/profile-wizard/steps/product-types/test/index.js (hunk 3 above)
Comment (psealock) [reply client/profile-wizard/steps/product-types/test/index.js:60]: Its a timing issue. If you remove the two `await` declarations from `onContinue` class method on `ProductTypes` it passes.

Hopefully, that leads you down the right path. Also, looking at the code, it appears `validateField` does not need to be an async method.

Comment (joshuatf) [reply client/profile-wizard/steps/product-types/test/index.js:60]: It does look like `validateField` does not need to be async and I handled some other clean-up work that seems to be related to wc-api in here, but unfortunately that does not get the tests passing for me.

If you log the mocked functions, it does not look like they are being passed at all to the component.  Even mocking `@wordpress/data` to override `dispatch` does not seem to override anything.  The real `updateProfileItems` gets called each time.

Comment (jeffstieler) [reply client/profile-wizard/steps/product-types/test/index.js:62]: > The real `updateProfileItems` gets called each time.

The component under test is wired to the stores. You can add an additional export in this file that is the component class before it gets wrapped in `compose()` to aid in testing.

See the PayPal payment task for an example:
https://github.com/woocommerce/woocommerce-admin/blob/aac0bde3caa0d94d7404e2e6a8918af3dd387ddb/client/task-list/tasks/payments/paypal.js#L21
https://github.com/woocommerce/woocommerce-admin/blob/aac0bde3caa0d94d7404e2e6a8918af3dd387ddb/client/task-list/test/payments.js#L12

Comment (joshuatf) [reply client/profile-wizard/steps/product-types/test/index.js:62]: Thanks, Jeff!

**Review by psealock — COMMENTED** [review]
Nice job getting those tests to pass.

I'm not sure how I got into this state, but if you aren't shown extensions to buy then it looks out of context:

Screen Shot 2020-08-27 at 1 38 17 PM

**Comment by joshuatf** [comment]
>I'm not sure how I got into this state, but if you aren't shown extensions to buy then it looks out of context:

@psealock Most likely a failed fetch to the WCCOM side to get the product data.  I think unrelated to this PR, but clearing that transient cache may resolve.  `wc_onboarding_product_data`

**Review by psealock — APPROVED** [review]
Nice job @joshuatf, testing well and code looks good :shipit:

**Pull request merged**